packages/router/src/admin/routes/GenerateGraphQlClient.route.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  ConduitBoolean,
+  ConduitError,
+  ConduitRoute,
+  ConduitRouteActions,
+  ConduitRouteParameters,
+  ConduitRouteReturnDefinition,
+  ConduitString,
+  TYPE,
+} from '@conduitplatform/commons';
+import { ConduitDefaultRouter } from '../..';
+import { generate } from '@graphql-codegen/cli';
+import path from 'path';
+import url from 'url';
+
+export function generateGraphQlClient(router: ConduitDefaultRouter) {
+  return new ConduitRoute(
+    {
+      path: '/router/generate-graphql-client',
+      action: ConduitRouteActions.POST,
+      bodyParams: {
+        plugin: ConduitString.Required,
+      },
+    },
+    new ConduitRouteReturnDefinition('generateGraphQlClient', {
+      response: TYPE.JSON,
+    }),
+    async (request: ConduitRouteParameters) => {
+      let response: any[] = [];
+      const outputPath = path.resolve(__dirname, 'dist/generated_graphql_client.d.ts');
+      try {
+        await generate({
+          schema: {
+            'http://localhost:3000/graphql': {
+              headers: {
+                clientid: request.headers.clientid as string,
+                clientsecret: request.headers.clientsecret as string,
+              },
+            },
+          },
+          generates: {
+            [outputPath]: {
+              plugins: selectPlugin(request.params!.plugin),
+            },
+          },
+        });
+        response.push({
```

Comment:
Whis is the response an array? You're returning a single object. Also the href won't work, the server doesn't know how to deliver files in this way. This is first and foremost a short-coming of Conduit. So you need to encode the file in base64 so that the user can download it. OR we need to do a new PR to add support for file downloads and implement this properly.

Reply:
How was this resolved?